### PR TITLE
LINK-1858 | authenticate with access code

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Makes it possible for strongly identified contact persons to read, update and delete their own signups and signup groups. In API side these actions are permitted if the contact person's access code and email match those of the signup's or signup group's contact person.

## Closes
[LINK-1858](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1858)

[LINK-1858]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1858?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by harriris-vincit** [comment]
Seems to be out-dated compared to the back-end changes: back-end now expects to use the one-time code when the user first does a GET request for a signup or group with the access_code parameter. After that, user linking is used instead of the parameter.

**Comment by jorilindell** [comment]
> Seems to be out-dated compared to the back-end changes: back-end now expects to use the one-time code when the user first does a GET request for a signup or group with the access_code parameter. After that, user linking is used instead of the parameter.

Very true. Removed the access code from the mutation queries.

File: src/domain/signupGroup/__tests__/utils.test.ts [hunk 1]
```diff
@@ -47,8 +48,48 @@ import {
   signupGroupPathBuilder,
   calculateTotalPrice,
   shouldCreatePayment,
+  canEditSignupGroup,
 } from '../utils';
 
+describe('canEditSignup function', () => {
```
Comment (harriris-vincit) [inline src/domain/signupGroup/__tests__/utils.test.ts:54]:
```suggestion
describe('canEditSignupGroup function', () => {
```

**Review by harriris-vincit — APPROVED** [review]
Looks good IMHO. Left a minor suggestion regarding suite description.

File: src/domain/signupGroup/__tests__/utils.test.ts (hunk 1 above)
Comment (jorilindell) [reply src/domain/signupGroup/__tests__/utils.test.ts]: Fixed

**Pull request merged**